test/mlir/conversion/onnx_to_tosa/NN/QuantizeLinear.mlir

```diff
@@ -0,0 +1,17 @@
+// RUN: onnx-mlir-opt --shape-inference --convert-onnx-to-tosa -cse %s -split-input-file | FileCheck %s
+
+func.func @test_quantizeLinear(%arg0 : tensor<32x3x224x224xf32>) -> tensor<32x3x224x224xi8> {
+  %0 = onnx.Constant dense<3.125000e-02> : tensor<f32>                       
+  %1 = onnx.Constant dense<0> : tensor<i8>                                   
+  %2 = "onnx.QuantizeLinear"(%arg0, %0, %1) {axis = 1 : si64} : (tensor<32x3x224x224xf32>, tensor<f32>, tensor<i8>) -> tensor<32x3x224x224xi8>
```

Comment:
I'm wondering if this PR also supports quantization to unsigned integers? Maybe add a test case for that?

Related: TOSA has this odd thing where it uses signless integers to say "signed" (like `i8`) but unsigned integers to say "unsigned" (like `ui8`). Is this the same in onnx-mlir?

Reply:
On paper it would probably work, but we don't have handling of the `ui8` type yet to my knowledge - I tried to add a test case according to @cmcgirr-amd comment and realised it was failing because the constants cannot be lowered

Reply:
What exactly fails with ui8? Which pass does not handle ui8?

Would the ONNXQuantizeLinearOpLoweringToTOSA need to be changed to support is correctly? If yes, I suggest to add a check there and then not apply the transform.

Reply:
Can you also add the case where it fails so we can track this. We will want this fixed in the future.

Reply:
The cast op in MLIR does in theory allow ui8 and ui16, but only if you look into the type definitions. The comment in https://github.com/llvm/llvm-project/blob/19732a3eaa978f2c65c233a97593b09e43a618cc/mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td#L1743 does not mention it and the specification https://www.mlplatform.org/tosa/tosa_spec.html#_cast does not list it as supported.

Reply:
Could we abuse the implementation of TOSA in MLIR for now to support unsigned? And push for the specification to change to support this? I would imagine this would also come with the push to support signed and unsigned integers in TOSA.